### PR TITLE
Add a test for 8.2.3391: Resizing a window crashes with showbreak and breakindentopt

### PR DESCRIPTION
**Pull request by yegappan**

### CODE REVIEW

File: src/testdir/test_breakindent.vim [hunk 1]
```diff
@@ -831,4 +831,22 @@ func Test_breakindent20_list()
   call s:close_windows('set breakindent& briopt& linebreak& list& listchars& showbreak&')
 endfunc
 
+" The following used to crash Vim. This is fixed by 8.2.3391.
+" This is a regression introduced by 8.2.2903.
```
Comment (dpelle) [inline src/testdir/test_breakindent.vim:835]: Just confirming that the test caused vim-8.2.3390 to crash and latest vim-8.2.3394 does not crash on Linux x86_64.
Thanks.

**Comment by vim-ml** [comment]
Hi,

On Wed, Sep 1, 2021 at 9:55 PM Dominique Pellé ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/testdir/test_breakindent.vim
> <https://github.com/vim/vim/pull/8828#discussion_r700743830>:
>
> > @@ -831,4 +831,22 @@ func Test_breakindent20_list()
>    call s:close_windows('set breakindent& briopt& linebreak& list& listchars& showbreak&')
>  endfunc
>
> +" The following used to crash Vim. This is fixed by 8.2.3391.
> +" This is a regression introduced by 8.2.2903.
>
> Just confirming that the test caused vim-8.2.3390 to crash and latest
> vim-8.2.3394 does not crash on Linux x86_64.
> Thanks.
>
>
>
Yes. The test crashes Vim 8.2.3390 and doesn't crash with 8.2.3394.

- Yegappan

**Comment by chrisbra** [comment]
thanks!

**Comment by vim-ml** [comment]
Hi Bram,

On Thu, Sep 2, 2021 at 11:05 AM Bram Moolenaar ***@***.***>
wrote:

> Thanks for creating this test.
> There is another check for n_extra becoming negative, and if I comment it
> out no test fails.
> Perhaps you can manage to write a test for that one too?
> This is in line 1215 of src/drawline.c
>
>
> I created a test that exercises this line. But I ran into some bugs.
If you source the following script:

-------------------------------------------------------------------------------------
10new
20vnew
call setline(1, repeat('aa ', 200))
setlocal breakindent breakindentopt=shift:4 linebreak showbreak=>>
setlocal number
set cpo+=n
normal 12gj
redraw!
normal $
redraw!
 -------------------------------------------------------------------------------------

You will see that the cursor is positioned a few characters before the end
of the line and it cannot be moved to the end of the line. Also, if you move
the cursor to the left by pressing "h" quite a few times (to cause the
screen
to scroll down), you will see the cursor can be positioned on the
'showbreak' characters.

- Yegappan

**Comment by vim-ml** [comment]
> > Thanks for creating this test.
> > There is another check for n_extra becoming negative, and if I comment it
> > out no test fails.
> > Perhaps you can manage to write a test for that one too?
> > This is in line 1215 of src/drawline.c
> 
> 
> I created a test that exercises this line. But I ran into some bugs.
> If you source the following script:
> 
> -------------------------------------------------------------------------------------
> 10new
> 20vnew
> call setline(1, repeat('aa ', 200))
> setlocal breakindent breakindentopt=shift:4 linebreak showbreak=>>
> setlocal number
> set cpo+=n
> normal 12gj
> redraw!
> normal $
> redraw!
>  -------------------------------------------------------------------------------------
> 
> You will see that the cursor is positioned a few characters before the end
> of the line and it cannot be moved to the end of the line. Also, if you move
> the cursor to the left by pressing "h" quite a few times (to cause the
> screen
> to scroll down), you will see the cursor can be positioned on the
> 'showbreak' characters.

It looks like this happens because the " >>" in the first line is not
counted, thus when w_skipcol is non-zero.  It's also strange that there
is a space before ">>".  Perhaps Christian knows what needs to change to
fix this.

-- 
The budget process was invented by an alien race of sadistic beings who
resemble large cats.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Pull request closed without merging**